lib/ansible/modules/web_infrastructure/bugzilla_bug.py

```diff
@@ -0,0 +1,160 @@
+        required: false
+'''
+
+
+EXAMPLES = '''
+
+# For fetching details about non-private bug
+- name: Get status for BZ 1429535
+  bugzilla_bug:
+    url: "https://bugzilla.redhat.com"
+    bug_id: 1429535
+  register: bug
+
+# For fetching details about private bug
+- name: Get status for BZ 14295
+  bugzilla_bug:
+    url: "https://bugzilla.redhat.com"
+    bug_id: 14295
+    url_username: userfoo
+    url_password: passwordfoo
+  register: bug
+
+# Register bug holds all bug details it can be access as below
+# For fetching status
+- debug: var=bug.bug.result.bugs[0].status
+# For fetching resolution
+- debug: var=bug.bug.result.bugs[0].resolution
+# For fetching severity
+- debug: var=bug.bug.result.bugs[0].severity
+'''
+
+RETURN = '''
+bug:
+  description: Fetches details of bug
+  returned: Bug Details
+  type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url, url_argument_spec
+from ansible.module_utils.json_utils import json
+
+
+def bz_get(module, url, data, method):
+    """ Helper function which will only do one thing that is
+    calling bugzilla server using fetch_url """
+    headers = {'Content-Type': 'application/json'}
+    url = url + "/jsonrpc.cgi?" + data
+    resp, info = fetch_url(module, url, headers=headers, method=method)
+    return resp, info
+
+
+def main():
+    argument_spec = url_argument_spec()
+    argument_spec.update(
+        url=dict(required=True, type='str'),
+        bug_id=dict(required=True, type='str')
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec
+    )
+
+    bug_id = module.params['bug_id']
+    user = module.params['url_username']
+    password = module.params['url_password']
+    url = module.params['url']
+
+    # Update module parameters by user's parameters if defined
+    if 'params' in module.params and isinstance(module.params['params'], dict):
+        module.params.update(module.params['params'])
+    # Remove the params
+    module.params.pop('params', None)
+
+    # Constructing the data/parameter for jsonrpc
+    if user and password:
+        data = 'params=[{"Bugzilla_login":"%s","Bugzilla_password":"%s",' \
+               '"ids":[%s]}]' % (user, password, bug_id)
+    else:
+        data = 'params=[{"ids":[%s]}]' % (bug_id)
+
+    # Combining method and data
+    data = "method=Bug.get&%s" % (data)
+    resp, info = bz_get(module, url, data, 'GET')
+
+    if info.get("status") != 200:
+        module.fail_json(msg="HTTP error " +
+                             str(info["status"]) + " " + info["msg"])
+
+    response = resp.read()
+    resp_json = json.loads(response)
+    error = resp_json.get("error")
+
+    # Addressing failure scenarios
+    # when username password is incorrect or
+    # accessing a private bug without credentials
+
+    if error is not None:
+        module.fail_json(msg=error)
+
+    bug_detail = resp_json
+    module.exit_json(bug=bug_detail)
```

Comment:
when I am fetching the bug details I am not making changes, so changed will remain false right ?